### DISCUSSION TITLE
[Messenger] Why all 3 dependencies?

### DISCUSSION THREAD

**Discussion by ghost** [symfony/symfony#43856] [Q&A] [answered]
Why does symfony/messenger has ampq-messenger, doctrine-messenger and redis-messenger as dependencies? If one uses doctrine transport there is not need to require ampq and redis transports. Shouldn't the desired transport package be required manually by a developer as another dependency in composer.json?

**Comment by W0rma** [comment, marked as answer]
The dependencies are included in symfony 5.1-5.4 to keep BC. The new packages are the result of https://github.com/symfony/symfony/pull/35422 which was released in symfony 5.1.

In symfony 6 they will not be included by default anymore (see https://github.com/symfony/symfony/blob/6.0/src/Symfony/Component/Messenger/composer.json).

**Reply by ghost** [reply]
Thank you.
